playback/src/audio_backend/gstreamer.rs

```diff
@@ -0,0 +1,115 @@
+use super::{Open, Sink};
+use gst::prelude::*;
+use gst::*;
+use std::sync::mpsc::{sync_channel, SyncSender};
+use std::{io, thread};
+use zerocopy::*;
+
+#[allow(dead_code)]
+pub struct GstreamerSink {
+    tx: SyncSender<Vec<u8>>,
+    pipeline: gst::Pipeline,
+}
+
+impl Open for GstreamerSink {
+    fn open(device: Option<String>) -> GstreamerSink {
+        gst::init().expect("Failed to init gstreamer!");
+        let pipeline_str_preamble = r#"appsrc caps="audio/x-raw,format=S16LE,layout=interleaved,channels=2,rate=44100" block=true max-bytes=4096 name=appsrc0 "#;
+        let pipeline_str_rest = r#" ! audioconvert ! autoaudiosink"#;
+        let pipeline_str: String = match device {
+            Some(x) => format!("{}{}", pipeline_str_preamble, x),
+            None => format!("{}{}", pipeline_str_preamble, pipeline_str_rest),
+        };
+        println!("Pipeline: {}", pipeline_str);
+
+        gst::init().unwrap();
+        let pipelinee = gst::parse_launch(&*pipeline_str).expect("New Pipeline error");
+        let pipeline = pipelinee
+            .dynamic_cast::<gst::Pipeline>()
+            .expect("Couldn't cast pipeline element at runtime!");
+        let bus = pipeline.get_bus().expect("Couldn't get bus from pipeline");
+        let mainloop = glib::MainLoop::new(None, false);
+        let appsrce: gst::Element = pipeline
+            .get_by_name("appsrc0")
+            .expect("Couldn't get appsrc from pipeline");
+        let appsrc: gst_app::AppSrc = appsrce
+            .dynamic_cast::<gst_app::AppSrc>()
+            .expect("Couldn't cast AppSrc element at runtime!");
+        let bufferpool = gst::BufferPool::new();
+        let appsrc_caps = appsrc.get_caps().expect("get appsrc caps failed");
```

Comment:
Pedantic, but could this be "Couldn't get appsrc caps" just to be consistent with the other messages?